utils/utils.go

```diff
@@ -51,8 +51,29 @@ func Remove(slice []string, s string) (result []string) {
 	return
 }
 
+func MapItems(source []string, transform func(string) string) []string {
+	target := make([]string, len(source))
+	for i := 0; i < len(source); i += 1 {
+		target[i] = transform(source[i])
+	}
+	return target
+}
+
+func Retry(attempts int, sleep time.Duration, f func() error) error {
+	var err error
+	for i := 0; i < attempts; i++ {
+		if err = f(); err == nil {
+			return nil
+		}
+		time.Sleep(sleep)
+		log.Println("retrying after error:", err)
+	}
+	return fmt.Errorf("Failed after %d retries. Last error: %v", attempts, err)
+
+}
+
 // AddLabel add a label to a resource metadata
-func AddLabel(obj metav1.Object, key string, value string) {
+func AddLabel(obj client.Object, key string, value string) {
```

Comment:
- why `client.Object` if we are only going to touch `meta` on the object but not any `runtime`?

Reply:
This is the new best practice for all `client` and `ctrlutil` functionality. They used to have segregation between functions that take `metav1.Object` and `runtime.Object` but then they introduced the `client.Object` which aggregates both. We are just following their footsteps

Reply:
I don't remember how many operator SDK versions / kubebuilder versions ago they did the change where they refactor all of the APIs from `runtime.Object` and `meta.Object` to `client.Object` for their own controller utils. You can check the their history their release nodes for more information